packages/@aws-cdk/aws-sqs/lib/queue-ref.ts

```diff
@@ -110,6 +110,93 @@ export abstract class QueueRef extends cdk.Construct implements s3n.IBucketNotif
       dependencies: [ this.policy! ]
     };
   }
+
+  /**
+   * Grant permissions to consume messages from a queue
+   *
+   * This will grant the following permissions:
+   *
+   *   - sqs:ChangeMessageVisibility
+   *   - sqs:ChangeMessageVisibilityBatch
+   *   - sqs:DeleteMessage
+   *   - sqs:ReceiveMessage
+   *   - sqs:DeleteMessageBatch
+   *   - sqs:GetQueueAttributes
+   *   - sqs:GetQueueUrl
+   *
+   * @param identity Principal to grant consume rights to
+   * @param queueActions additional queue actions to allow
+   */
+  public grantConsumeMessages(identity?: iam.IPrincipal, ...queueActions: string[]) {
```

Comment:
I don't like this pattern where you add in more actions at the end. It's inconsistent with other instances of this API, and also unnecessary. If you really want this then you can call `grant()` directly.

Reply:
Good point. Fixed.